PSGSuite/Public/Users/Invoke-GSUserOffboarding.ps1

```diff
@@ -0,0 +1,158 @@
+        [string]
+        $License = "Google-Vault-Former-Employee"
+    )
+    Begin {
+        function New-RandomPassword {
+            Param (
+                [parameter(Mandatory = $false)]
+                [int]
+                $Length = 15
+            )
+            $ascii = $null
+            for ($a = 33;$a –le 126;$a++) {
+                $ascii += ,[char][byte]$a
+            }
+            for ($loop = 1; $loop –le $length; $loop++) {
+                $randomPassword += ($ascii | Get-Random)
+            }
+            return ([String]$randomPassword)
+        }
+    }
+    Process {
+        foreach ($U in $User) {
+            Resolve-Email ([ref]$U)
+            if ($PSCmdlet.ShouldProcess("Offboarding user: $U")) {
+                Write-Verbose "Offboarding user: $U"
+                "[$(Get-Date -Format o)] Starting offboard of user: $U"
+                $_user = @{User = $U}
+                $updateParams = @{Confirm = $false}
+                foreach ($opt in $options) {
+                    switch -RegEx ($opt) {
+                        '(Full|Suspend)' {
+                            $updateParams['Suspended'] = $true
+                        }
+                        '(Full|SetRandomPassword)' {
+                            $updateParams['Password'] = ConvertTo-SecureString (New-RandomPassword) -AsPlainText -Force
+                            $updateParams['ChangePasswordAtNextLogin'] = $true
+                        }
+                        '(Full|MoveToOrgUnit)' {
+                            if ($PSBoundParameters.ContainsKey('DestinationOrgUnit')) {
+                                $updateParams['OrgUnitPath'] = $PSBoundParameters['DestinationOrgUnit']
+                            }
+                            else {
+                                throw "No DestinationOrgUnit provided!! Stopping further processing"
+                                exit 1
+                            }
+                        }
+                    }
+                }
+                "[$(Get-Date -Format o)] [$U] Updating user"
+                Update-GSUser @_user @updateParams | Format-List PrimaryEmail,@{N = "FullName";E = {$_.name.fullName}},Suspended,ChangePasswordAtNextLogin,OrgUnitPath
+                if ($Options -contains 'Full' -or $Options -contains 'ClearASPs') {
+                    "[$(Get-Date -Format o)] [$U] Retrieving App Specific Passwords to remove"
+                    $ASPs = Get-GSUserASPList @_user
+                    if ($ASPs) {
+                        foreach ($ASP in $ASPs) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The cmdlet 'Invoke-GSUserOffboarding' returns an object of type 'System.String' but this type is not declared in the OutputType attribute.](https://app.codacy.com/manual/scrthq/PSGSuite/pullRequest?prid=4742353)